### ISSUE TITLE
Potential issue with CommentedOutCode

### ISSUE THREAD

**Issue by mboynes** [squizlabs/PHP_CodeSniffer#862]
I ran into an issue today where the following line of code was causing an infinite loop:

```php
// Base config function.
```

The rule that was getting stuck, as you'd probably expect, was `Squiz.PHP.CommentedOutCode`. I haven't looked into it, but it seems like the word "function" cannot appear in a comment without being followed by an actual function definition (e.g. `// Base config function demo(){}` does not trigger an infinite loop, but any iteration with the word "function" does).

Probably related to #826.

**Comment by gsherwood** [comment]
I'm not getting any infinite loops on this code in the stable 2.5.0 release, and can confirm the commented out code sniff is working fine. But I do get an issue on the latest code from the master branch, but that is in the tokenizer itself. I'm looking into the tokenizer now.

**Comment by gsherwood** [comment]
Should clarify: the commented out code sniff makes use of the tokenizer, which is why it appears to be getting stuck in there. But it's a core tokenizer error in this case.

**Comment by gsherwood** [comment]
Yep, was a tokenizer error, and is fixed now. I can't see how this could have been generating errors in the released version as it was introduced by the commit [ https://github.com/squizlabs/PHP_CodeSniffer/commit/ee0f59ce852d1498afa8c580d392c14e0e58a7f6 ], so I've left it out of the changelog.

Thanks a lot for reporting it though. There was no unit test to catch it, so it would have slipped through into the next release.

**Comment by gsherwood** [comment]
I put the wrong bug ID in the commit message. The commit is here: https://github.com/squizlabs/PHP_CodeSniffer/commit/a81b2e693f7b371a5eb43735047f34c22e2acbe0

**Comment by mboynes** [comment]
Thanks, I appreciate you tracking that down and patching it so quickly!
